legate/core/store.py

```diff
@@ -246,13 +254,17 @@ def detach_external_allocation(
         assert self.attached_alloc is not None
         detach = attachment_manager.remove_detachment(self.detach_key)
         detach.unordered = unordered  # type: ignore[union-attr]
-        attachment_manager.detach_external_allocation(
+        detach_future = attachment_manager.detach_external_allocation(
             self.attached_alloc, detach, defer
```

Comment:
Pass `self.field` here (see further discussion on this on another comment).

```suggestion
            self.attached_alloc, detach, defer, self.field
```